src/Raven.Server/Documents/AI/Embeddings/IEmbeddingValue.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.IO;
+
+namespace Raven.Server.Documents.AI.Embeddings;
+
+public interface IEmbeddingValue
+{
+    ReadOnlySpan<byte> GetEmbedding();
+    
+    Stream GetEmbeddingStream();
```

Comment:
Case in point for why I don't like this.
The tests for this are broken, see here:
https://github.com/Lwiel/ravendb/blob/b6be9d7c4617e43316192f9811a7c35996cfe2cf/test/SlowTests/Server/Documents/AI/Embeddings/Cache/QueryEmbeddingsCacherTests.cs#L48

We are sending an embedding with 3 floats, but we store 3 _bytes_ in the vector.

Reply:
We are aware that they are broken, they started failing after Friday rewrite of the query embedding worker

Reply:
Fixing that are removing `IEmbeddingValue` in favor of `ReadOnlyMemory` now